appium/sample-scripts/python/testdroid_android_hybrid.py

```diff
@@ -6,51 +6,100 @@
 import os
 import time
 import unittest
-from device_finder import DeviceFinder
+import subprocess
 from time import sleep
-from selenium import webdriver
+from appium import webdriver
+from device_finder import DeviceFinder
+from selenium.common.exceptions import WebDriverException
+from selenium.common.exceptions import NoSuchElementException
+from appium.common.exceptions import NoSuchContextException
 
 def log(msg):
     print (time.strftime("%H:%M:%S") + ": " + msg)
 
 class TestdroidAndroid(unittest.TestCase):
 
+    """
+    Take screenshot and store files to defined location, with numbering prefix
+
+    :Args:
+    - name - files are stored as #_name
+    """
+
     def screenshot(self, name):
-        screenshotName = str(self.screenshot_count) + "_" + name + ".png" 
-        log ("Taking screenshot: " + screenshotName)
-        sleep(1) # wait for animations to complete before taking screenshot
-        self.driver.save_screenshot(self.screenshot_dir + "/" + screenshotName)
+        screenshot_name = str(self.screenshot_count) + "_" + name + ".png"
+        log("Taking screenshot: " + screenshot_name)
+        # on Android, switching context to NATIVE_APP for screenshot
+        # taking to get screenshots also stored to Testdroid Cloud
+        # device run view. After screenshot switching back to WEBVIEW.
+        if not "NATIVE_APP" in self.driver.current_context:
+            self.driver.switch_to.context("NATIVE_APP")
+            self.driver.save_screenshot(self.screenshot_dir + "/" + screenshot_name)
+            contexts = self.driver.contexts
+            self.driver.switch_to.context(self.driver.contexts[-1])
+        else:
+            self.driver.save_screenshot(self.screenshot_dir + "/" + screenshot_name)
+
```

Comment:
Onko tarpeellista if not "NATIVE_APP" in ... tarpeellinen? Toimii vaikka aina tehtäisiin kontekstin vaihto.